Hdfs/Zookeeper & Oracle/Scientific

shurl opened the pull request:
This adds support for the HDFS and Zookeeper plugins. 

Also adds support for Oracle and Scientific distros.

carlosperello commented on README.md:
```diff
@@ -24,22 +25,24 @@ This cookbook installs the v2 [Server Density](http://www.serverdensity.com/) ag
 - Debian
 - RHEL
 - CentOS
+- Amazon
```
Where is Oracle and Scientific?

shurl replied:
Added!

carlosperello commented on recipes/default.rb:
```diff
@@ -186,6 +186,32 @@
+                  )
+        notifies :restart, 'service[sd-agent]', :delayed
+    end
+    if node['serverdensity']['hdfs_namenode_jmx_uri']
```
Bad indentation ?

shurl commented:
@carlosperello I've addressed the issues you pointed out. 

The bad indentation was intentional, but after chatting yesterday I've rearranged the logic. It doesn't appear to attempt the install on the second instance when testing: 

``` 
* yum_package[sd-agent-hdfs] action install
   - install version 2.1.4-1.el7.centos of package sd-agent-hdfs
* template[/etc/sd-agent/conf.d/hdfs_datanode.yaml] action create (up to date)
* yum_package[sd-agent-hdfs] action install (up to date)
* template[/etc/sd-agent/conf.d/hdfs_namenode.yaml] action create (up to date)
* yum_package[sd-agent-varnish] action install
```

carlosperello commented:
👍  will merge on Monday.
